performance/api_load.go

```diff
@@ -0,0 +1,144 @@
+	"encoding/base64"
+	"github.com/smartcontractkit/chainlink/internal/cltest"
+	"github.com/tsenart/vegeta/lib"
+	"fmt"
+	"time"
+	"log"
+	"github.com/smartcontractkit/chainlink/store"
+	"net/http"
+)
+
+var (
+	schemaPrefix = "../internal/fixtures/web/"
+	schemas      = [...]string{
+		"hello_world_job.json",
+		"uint256_job.json",
+	}
+)
+
+// Pre-load the schemas prior to run-time to reduce overhead
+func GetSchemas() (schemaBytes [][]byte) {
+	for _, schema := range schemas {
+		byteArray, err := ioutil.ReadFile(schemaPrefix + schema)
+		if err != nil {
+			log.Fatal(err)
+		}
+		schemaBytes = append(schemaBytes, byteArray)
+	}
+	return
+}
+
+// Get the basic auth http header
+func GetBasicAuthHeader(store *store.Store) (header *http.Header) {
+	header = &http.Header{}
+	authString := store.Config.BasicAuthUsername + ":" + store.Config.BasicAuthPassword
+	header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(authString)))
+	return
+}
+
+// Create the vegata targets based on the schemas used for creating jobs
+func GetPostJobTargets(app *cltest.TestApplication) (targets []vegeta.Target) {
+	schemas := GetSchemas()
+	for _, schema := range schemas {
+		targets = append(targets, vegeta.Target{
+			Method: "POST",
+			URL:    fmt.Sprintf("%s/v2/jobs", app.Server.URL),
+			Body:   schema,
+			Header: *GetBasicAuthHeader(app.Store),
+		})
+	}
+	return
+}
+
+// Create the vegata targets for viewing the jobs created
+func GetViewJobTargets(app *cltest.TestApplication) (targets []vegeta.Target) {
+	jobs, err := app.Store.Jobs()
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, job := range jobs {
+		targets = append(targets, vegeta.Target{
+			Method: "GET",
+			URL:    fmt.Sprintf("%s/v2/jobs/%s", app.Server.URL, job.ID),
+			Header: *GetBasicAuthHeader(app.Store),
+		})
+	}
+	return
+}
+
+// Create the vegata targets for creating job runs
+func GetJobRunTargets(app *cltest.TestApplication) (targets []vegeta.Target) {
+	jobs, err := app.Store.Jobs()
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, job := range jobs {
+		targets = append(targets, vegeta.Target{
+			Method: "POST",
+			URL:    fmt.Sprintf("%s/v2/jobs/%s/runs", app.Server.URL, job.ID),
+			Header: *GetBasicAuthHeader(app.Store),
+		})
+	}
+	return
+}
+
+// Create the vegata targets for viewing job runs
+func GetViewJobRunTargets(app *cltest.TestApplication) (targets []vegeta.Target) {
+	jobs, err := app.Store.Jobs()
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, job := range jobs {
+		targets = append(targets, vegeta.Target{
+			Method: "GET",
+			URL:    fmt.Sprintf("%s/v2/jobs/%s/runs", app.Server.URL, job.ID),
+			Header: *GetBasicAuthHeader(app.Store),
+		})
+	}
+	return
+}
+
+// Wait for all the jobs and their tasks to complete after the job runs
+func WaitForJobRunsToComplete(app *cltest.TestApplication) (time.Duration, error) {
+	timeout := float64(60)
+	jobs, err := app.Store.Jobs()
+	if err != nil {
+		log.Fatal(err)
+	}
+	start := time.Now()
+	for {
```

Comment:
Yeah, definitely a concern. I commented out the checking of this for anything above 20 requests per second because it was doing that. I couldn't find any other way to achieve the same result though. Is there any timestamps which provide a time a task/job was completed rather than when the run was created?